feat: add OpenRouter AI provider support

lkonga opened the pull request:
Adds support for openrouter.ai as a new provider, allowing users to access various AI models

jnsahaj commented:
Appreciate the addition :)
I'll merge it once I get back to my laptop in a few days

jnsahaj commented on README.md:
Are only these 2 models supported?

lkonga commented on README.md:
every openai and anthropic model is supported

https://openrouter.ai/models?q=anthropic
https://openrouter.ai/models?q=openai

jnsahaj commented on README.md:
In that case I think it will be better to provide a link to see the list of all available models

jnsahaj commented:
Hey, can you rebase your PR with main and resolve the conflicts?

lkonga commented:
Done 👍

jnsahaj commented on README.md:
Maybe I'm missing something since I don't personally use OpenRouter, but why are we only listing Anthropic and OpenAI models?
Why can't the list URL be `https://openrouter.ai/models`?

lkonga commented on README.md:
Using existing provider models is more stable, but the OpenRouter models page works well too, I'm getting good results with qwen, though other models may vary

jnsahaj commented:
Can you share the output of these commands to verify that the change is working as expected
```sh
cargo r -- -p "openrouter" -k "your-key" explain HEAD
cargo r -- -p "openrouter" -m "invalid-model" -k "your-key" explain HEAD
```

lkonga commented:
```bash
cargo r -- -p "openrouter" -k "sk-or-v1-xxxx...xxxx" explain HEAD
```
```
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.12s
     Running `target/debug/lumen -p openrouter -k sk-or-v1-xxxx...xxxx explain HEAD`
┄Entity: Commit

commit c20307eb1b95d05d41d867e40f43c371889e6e52 | lkonga 2558702+lkonga@users.noreply.github.com | 2024-11-16 23:02:16

┄┄docs: update OpenRouter model information in README

✓ Done
Core Changes:

• Simplified the OpenRouter model reference in README by replacing specific model listings with a single link to all
  models
• Changed from separate Anthropic/OpenAI model links to a unified model list

Impact:

• More maintainable documentation as it doesn't need updates when specific models change
• Users now have a cleaner, single reference point to view all available OpenRouter models
```

```bash
cargo r -- -p "openrouter" -m "invalid-model" -k "sk-or-v1-xxxx...xxxx" explain HEAD
```

```
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/lumen -p openrouter -m invalid-model -k sk-or-v1-xxxx...xxxx explain HEAD`
┄Entity: Commit

commit c20307eb1b95d05d41d867e40f43c371889e6e52 | lkonga 2558702+lkonga@users.noreply.github.com | 2024-11-16 23:02:16

┄┄docs: update OpenRouter model information in README

error: API request failed with status code 400 Bad Request: invalid-model is not a valid model ID
```
